### ISSUE TITLE
Fix Airspace issue

### ISSUE THREAD

**Issue by jeremyVignelles** [ZeBobo5/Vlc.DotNet#296]
Many users have reported Airspace issues while using this library. One issue to rule them all and in the darkness bind them.

Sample code (taken from #41): 
```
<Window
        xmlns="http://schemas.microsoft.com/winfx/2006/xaml/presentation"
        xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
        xmlns:Wpf="clr-namespace:Vlc.DotNet.Wpf;assembly=Vlc.DotNet.Wpf" x:Class="LivestreamViewer.MainWindow"
        Title="MainWindow" Height="449" Width="753">
    <Grid>
        <Wpf:VlcControl x:Name="vlc" Margin="0,0,0,13" Panel.ZIndex="-1"/>
        <Grid x:Name="overlayGrid" HorizontalAlignment="Stretch" Height="25" Margin="0,0,0,0" VerticalAlignment="Bottom">
            <Rectangle Fill="Black" HorizontalAlignment="Stretch" Margin="0,0,0,0" Stroke="Black" VerticalAlignment="Stretch" IsHitTestVisible="False" Opacity="0.5"/>
        </Grid>
    </Grid>
</Window>
```

Technical explanations: 
- https://msdn.microsoft.com/en-us/library/aa970688(v=vs.85).aspx
- https://blogs.msdn.microsoft.com/dwayneneed/2013/02/26/mitigating-airspace-issues-in-wpf-applications/
- https://msdn.microsoft.com/en-us/library/ms742522(v=vs.85).aspx

Following issues are impacted (and will be closed):
- #8
- #12 
- #41 
- #143

Proposed workarounds/hacks (untested):
- https://stackoverflow.com/questions/5978917/render-wpf-control-on-top-of-windowsformshost/23662410#23662410
- https://github.com/kolorowezworki/Airhack (thanks @kolorowezworki)

**Comment by kolorowezworki** [comment]
https://github.com/kolorowezworki/Airhack
Try this. It's quite dirty, but it works ;)

**Comment by jeremyVignelles** [comment]
Thanks for providing a workaround! I'll keep this open until this is fixed or considered as an acceptable solution (I'd prefer the first option)

**Comment by kolorowezworki** [comment]
We cannot fix airspace. Maybe we can fix VLC control, but not general problem with Winforms. This is caused by the method of rendering WPF and Winforms controls and windows. I think, at the moment, we can base only on workarounds.

**Comment by jeremyVignelles** [comment]
This could be fixed by #249 : If there is no reference to winforms, then there is no airspace issue, am I wrong? the option I'm thinking of is using DirectX to do the rendering, rather than using WinForm/HWND. vlc-winrt is already using that technique for win10 apps

**Comment by kolorowezworki** [comment]
Sure. No Winforms/Win32Api == no airspace. DirectX is rendered the similar way to WPF. Avoiding problem is avoid, not solve ;) #249 is not about fixing airspace, rather about making something very new and different of current solution.

**Comment by kolorowezworki** [comment]
We could use HWND of WPF Window directly and it will not need a reference to Windows.Forms, but airspace effect will not disappear.

**Comment by jeremyVignelles** [comment]
A comment in #257 shows a workaround for the Airspace issue in stackoverflow : https://stackoverflow.com/questions/5978917/render-wpf-control-on-top-of-windowsformshost/23662410#23662410

**Comment by JUSTHassan** [comment]
A powerful workaround [https://stackoverflow.com/a/43695349/1576027](https://stackoverflow.com/a/43695349/1576027)

`<interop:AirspaceDecorator AirspaceMode="Redirect" Background="White" IsInputRedirectionEnabled="True" IsOutputRedirectionEnabled="True"> <Wpf:VlcControl x:Name="vlcControl" Margin="0,0,0,0"/>
</interop:AirspaceDecorator>`

**Comment by jeremyVignelles** [comment]
Will be solved in 3.0.0 when #365 gets merged

**Comment by jeremyVignelles** [comment]
Should be fixed in 3.0.0-develop250
